cmd/opentelemetry/app/internal/esclient/client.go

```diff
@@ -172,20 +175,20 @@ func NewElasticsearchClient(params config.Configuration, logger *zap.Logger) (El
 	if err != nil {
 		return nil, err
 	}
-	if params.GetVersion() == 0 {
+	esVersion := int(params.GetVersion())
+	if esVersion == 0 {
 		esPing := elasticsearchPing{
 			username:     params.Username,
 			password:     params.Password,
 			roundTripper: roundTripper,
 		}
-		esVersion, err := esPing.getVersion(params.Servers[0])
+		esVersion, err = esPing.getVersion(params.Servers[0])
```

Comment:
Does it make sense to check length of `.Servers` before dereferencing?  It does appear that right now it's impossible to pass Servers with length 0 into this method, but the code that prevents that is fairly distant from this.

Might be better if this method protects itself.

Reply:
+1 I am will submit a follow-up PR to fix that

Reply:
https://github.com/jaegertracing/jaeger/pull/2411